src/Microsoft.OData.Client/DataServiceContext.cs

```diff
@@ -3899,21 +3948,55 @@ private void UpdateObjectInternal(object entity, bool failIfNotUnchanged)
                 return;
             }
 
+            if(dependsOnIds != null)
+            {
+                //In DataServiceContext.UpdateObject, we pass entity Id(s) as dependsOnIds.
+                //We will extract ChangeOrder from the entity descriptor based on the Id.
+                //This is because we pass the ChangeOrder as Content-ID to the batch writer.
+                //We will pass the Content-ID(s) as dependsOnIds to the batch writer.
+                List<string> dependsOnIdsAsChangeOrders;
+                HashSet<string> dependsOnChangeSetIds;
+
+                GetDependsOnChangeOrdersAndChangeSetIds(
+                    dependsOnIds,
+                    this.EntityTracker.Entities,
+                    this.Model,
+                    out dependsOnIdsAsChangeOrders,
+                    out dependsOnChangeSetIds);
+                resource.DependsOnIds = dependsOnIdsAsChangeOrders;
+                resource.DependsOnChangeSetIds = dependsOnChangeSetIds.ToList();
+            }
+
             resource.State = EntityStates.Modified;
+            resource.ChangeSetId = Guid.NewGuid().ToString();

```

Comment:
@KenitoInc Is the assumption here that `ChangeSetId` was null? Would there be side effects if there was a value and it gets overwritten?

Reply:
Later found out that this is a property that you added...